pkg/pgmodel/ingestor/series_writer.go

```diff
@@ -235,7 +237,16 @@ func (h *seriesWriter) fillLabelIDs(ctx context.Context, infos map[string]*perMe
 			infoBatches = append(infoBatches, info)
 			items += len(namesSlice.Elements)
 		}
+		// since info.maxPos is only updated with fetched labels we need to iterate through cached as well
+		for _, cachedLabel := range info.cachedLabels {
+			if val, ok := labelMap[cachedLabel]; ok {
+				if int(val.Pos) > info.maxPos {
+					info.maxPos = int(val.Pos)
```

Comment:
Yeah we have to do it. Since some labels are cached and we need `maxPos` to be correct (meaning it contains the max for both fetched and cached labels)

Reply:
No, I meant, can you start calculating the max position when fetching cached entries, so you don't have to iterate again through the cached labels.